src/Compilers/CSharp/Portable/Emitter/Model/FieldSymbolAdapter.cs

```diff
@@ -186,7 +186,9 @@ bool Cci.IFieldDefinition.IsReadOnly
             get
             {
                 CheckDefinitionInvariant();
-                return this.IsReadOnly || (this.IsConst && !this.IsMetadataConstant);
+                return this.IsReadOnly ||
+                    (this.IsConst && !this.IsMetadataConstant) ||
+                    (this.AssociatedSymbol is PropertySymbol property && property.SetMethod?.IsInitOnly == true);
```

Comment:
>property.SetMethod?.IsInitOnly == true [](start = 73, length = 38)

Addition of this code feels unexpected. I would expect us to mark the field as readonly when we create it.